src/mainviews/create-job.tsx

```diff
@@ -113,11 +113,18 @@ export function CreateJob(props: ICreateJobProps): JSX.Element {
           envList[0].name
         )?.map(format => format.name);
 
+        // Does the default environment support time zones?
+        let newTimeZone = 'UTC';
+        if (!envList[0].utc_only) {
+          newTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+        }
```

Comment:
Ah, something else I noticed: this effect hook only gets called on initial render, since this is being used to fetch the initial environment list.

This means switching environments may not update the new timezone as expected.